src/main/Core/FileSystemUtility/NodeJsFileSystemUtility.test.ts

```diff
@@ -0,0 +1,325 @@
+import { mkdir, rm } from "fs";
+import { join } from "path";
+import { afterEach, beforeEach, describe, expect, it } from "vitest";
+import { NodeJsFileSystemUtility } from "./NodeJsFileSystemUtility";
+
+const createFolder = (path: string) =>
+    new Promise<void>((resolve, reject) => {
+        mkdir(path, (error) => {
+            if (error) {
+                reject(error);
+            } else {
+                resolve();
+            }
+        });
+    });
+
+const deleteFolder = (path: string) =>
+    new Promise<void>((resolve, reject) => {
+        rm(path, { recursive: true }, (error) => {
+            if (error) {
+                reject(error);
+            } else {
+                resolve();
+            }
+        });
+    });
```

Comment:
Why don't you use the promise based version?

Reply:
Hehe good point. I'll adjust it.

Reply:
Done.